
syntax = "proto3";
package renderPartition;

option java_package = "com.imageworks.spcue.grpc.renderpartition";
option java_multiple_files = true;


// -------- Services --------]

service RenderPartitionInterface {
    // Deletes the host local setup.  Any proc that is running from this should be killed or else the counts
    // could be off.
    rpc Delete(RenderPartDeleteRequest) returns (RenderPartDeleteResponse);

    // Reset the maximum amount of cores and memory for this render partition.
    rpc SetMaxResources(RenderPartSetMaxResourcesRequest) returns (RenderPartSetMaxResourcesResponse);
}


// -------- Enums --------]

enum RenderPartitionType {
    JOB_PARTITION = 0;
    LAYER_PARTITION = 1;
    FRAME_PARTITION = 2;
}


// -------- Primary Message Types --------]

message RenderPartition {
    string id = 1;
    string job = 2;
    string layer = 3;
    string frame = 4;
    RenderPartitionType render_part_type = 5;
    string host = 6;
    int32 cores = 7;
    int32 gpu = 8;
    int64 memory = 9;
    int32 max_cores = 10;
    int32 max_gpu = 11;
    int64 max_memory = 12;
    int64 max_gpu_memory = 13;
    int32 threads = 14;
}

message RenderPartitionSeq {
    repeated RenderPartition render_partitions = 1;
}


// -------- Requests & Responses --------]

message RenderPartDeleteRequest {
    RenderPartition render_partition = 1;
}

message RenderPartDeleteResponse {} // Empty

message RenderPartSetMaxResourcesRequest {
    RenderPartition render_partition = 1;
    int32 cores = 2;
    int32 gpu = 3;
    int64 memory = 4;
    int64 gpu_memory = 5;
}

message RenderPartSetMaxResourcesResponse {} // Empty
